app/controllers/incomeSources/add/UKPropertyBusinessAccountingMethodController.scala

```diff
@@ -0,0 +1,170 @@
+        if (isAgent) {
+          Future.successful(Redirect(controllers.incomeSources.add.routes.CheckUKPropertyDetailsController.showAgent())
+            .addingToSession(addUkPropertyAccountingMethod -> "cash"))
+        } else {
+          Future.successful(Redirect(controllers.incomeSources.add.routes.CheckUKPropertyDetailsController.show())
+            .addingToSession(addUkPropertyAccountingMethod -> "cash"))
+        }
+      } else {
+        Future.successful(Ok(view(
+          form = UKPropertyBusinessAccountingMethodForm.form,
+          postAction = postAction,
+          isAgent = isAgent,
+          backUrl = backUrl,
+          btaNavPartial = user.btaNavPartial
+        )(user, messages)))
+      }
+    } else {
+      Future.successful(Ok(customNotFoundErrorView()(user, messages)))
+    } recover {
+      case ex: Exception =>
+        Logger("application").error(s"${if (isAgent) "[Agent]"}" +
+          s"Error getting AddUkPropertyAccountingMethod page: ${ex.getMessage}")
+        errorHandler.showInternalServerError()
+    }
+  }
+
+  def shouldRedirectToCheckDetailsPage(userProperties: List[PropertyDetailsModel]): Boolean = {
+    if(userProperties.filter(_.isUkProperty).size > 0) {
+      true
+    } else {
+      false
+    }
+  }
+
+  def show(): Action[AnyContent] =
+    (checkSessionTimeout andThen authenticate andThen retrieveNino
+      andThen retrieveIncomeSources andThen retrieveBtaNavBar).async {
+      implicit user =>
+        handleRequest(
+          isAgent = false
+        )
+    }
+
+  def showAgent(): Action[AnyContent] = Authenticated.async {
+    implicit request =>
+      implicit user =>
+        getMtdItUserWithIncomeSources(incomeSourceDetailsService).flatMap {
+          implicit mtdItUser =>
+            handleRequest(
+              isAgent = true
+            )
+        }
+  }
+
+  def submit: Action[AnyContent] = (checkSessionTimeout andThen authenticate andThen retrieveNino
+    andThen retrieveIncomeSources andThen retrieveBtaNavBar).async {
+    implicit user =>
+      UKPropertyBusinessAccountingMethodForm.form.bindFromRequest().fold(
+        hasErrors => Future.successful(BadRequest(view(
+          form = hasErrors,
+          postAction = controllers.incomeSources.add.routes.UKPropertyBusinessAccountingMethodController.submit(),
+          backUrl = controllers.incomeSources.add.routes.CheckUKPropertyStartDateController.show().url,
+          isAgent = false
+        ))),
+        validatedInput => {
+          if (validatedInput.equals(Some("cash"))) {
+            Future.successful(Redirect(controllers.incomeSources.add.routes.CheckUKPropertyDetailsController.show())
+              .addingToSession(addUkPropertyAccountingMethod -> "cash"))
+          } else {
+            Future.successful(Redirect(controllers.incomeSources.add.routes.CheckUKPropertyDetailsController.show())
+              .addingToSession(addUkPropertyAccountingMethod -> "accruals"))
+          }
+        }
```

Comment:
Create a request handler for submit and submitAgent.. as the code is same and repeated.